src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -398,14 +515,28 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
 
             ODataNestedResourceInfo link = (ODataNestedResourceInfo)this.reader.Item;
 
-            MaterializerEntry entry;
-            ODataResourceSet feed;
-            if (this.TryReadFeedOrEntry(false, out feed, out entry))
+            MaterializerNestedEntry nestedEntry;
+
+            if (this.TryReadFeedOrEntry(false, out ODataResourceSet feed, out MaterializerDeltaFeed deltaFeed, out MaterializerEntry entry))
             {
                 if (feed != null)
                 {
                     MaterializerNavigationLink.CreateLink(link, feed, this.materializerContext);
                 }
+                else if (deltaFeed != null)
+                {
+                    nestedEntry = MaterializerNestedEntry.CreateNestedResourceInfo(
+                        link, 
+                        new List<IMaterializerState>(), 
+                        this.materializerContext);
+
+                    nestedEntry.AddNestedItem(deltaFeed);
+
+                    if (this.items?.Count > 0)

```

Comment:
I think you should add a `Debug.Assert(this.items != null, "yada yada");` here and eliminate the null propagation operator. If `this.items` is null at this point and you fail to push the nested entry into the stack, I don't think progressing under those circumstances makes sense

Reply:
the stack is only used when we are reading a delta payload. If we are just reading a single payload then this.items will be null hence the check.